### ISSUE TITLE
Enforce scraper output fields

### ISSUE THREAD

**Issue by tkjones117** [livgust/covid-vaccine-scrapers#156]
To make sure we have the write fields we need for writing to our FaunaDB, we should enforce that each scraper has certain outputs. 
We can use a test to do this.

@livgust  started a branch to do this a while back: `results-shape-test`
https://github.com/livgust/covid-vaccine-scrapers/compare/results-shape-test

For FaunaDB, I need the following fields: 
```
    name,
    street,
    city,
    zip,
    availability, # (needs to have correct object structure with hasAvailability, numberAvailableAppointments, and (optional) signUpLink
    hasAvailability,
    extraData, # (optional)
    timestamp,
    latitude,
    longitude,
    signUpLink, # (optional)
```
So this ticket would involve writing a test to make sure that each scraper outputs those fields.

**Comment by johnhawkinson** [comment]
In  #152, I added `siteTimestamp` as the optional time of the data as reported by the vendor's site.

**Comment by tkjones117** [comment]
@livgust is someone working on this rn? I see you moved to "review in progress". 
also this ticket may be duped by https://github.com/livgust/covid-vaccine-scrapers/issues/79

This will help unblock FaunaDB work so it would be good to get someone on it if no one is

**Comment by tkjones117** [comment]
bump @livgust and also @harcod. Rob are you working on this?

**Comment by tkjones117** [comment]
bump @livgust, wondering the status on this. I assume no one is working on it?

**Comment by tkjones117** [comment]
After some discussion with @livgust today, this is not a blocker for moving forward with the DB migration. It's a nice-to-have regardless of using S3/Fauna. But we can proceed with Fauna without being 100% sure that our scraper outputs match a certain structure, since that's what we do with S3 publishing as-is.

**Comment by pfurbacher** [comment]
I'm wondering if the `readme.md` should have a section added which will provide this list to new developers? Something like the following inserted right after the bullet point about using puppeteer: 
***

A scraper may return data for a single site, or an array of data in cases where multiple sites are powered by the same vendor (e.g., Color, MyChart, ZocDoc). For each site, the scraper should return an Object containing the following fields:
```
    name,
    street,
    city,
    zip,
    availability, # (see notes below for structure)
    hasAvailability,
    extraData, # (optional, see notes below)
    timestamp,
    latitude,
    longitude,
    signUpLink, # (optional: if a site has availability, include the signUpLink)
```
The `availability` field must look like this: 
```
    availability: {
        "<date>": {
            numberAvailableAppointments: { } // see note below on structure
            hasAvailability: // set it to true if numberAvailableAppointments > 0), 
            signUpLink: (optional: the URL for that particular date; not applicable to all sites ) 
        }, ... // repeated for each date with availability
    }
```

The `date` property name looks like this: `"3/30/2021"`.
 
The `numberAvailableAppointments` object is either empty (`{}`) or contain properties describing availability by date. `numberAvailableAppointments` must never be `null` or `undefined`.

Here's an example of a site with availability on three days: 
```
{
  "3/30/2021": {
    numberAvailableAppointments: 2,
    hasAvailability: true,
  },
  "3/31/2021": {
    numberAvailableAppointments: 3,
    hasAvailability: true,
  },
  "4/2/2021": {
    numberAvailableAppointments: 4,
    hasAvailability: true,
  },
}
```

If a scraper returns more than a single site's results, it must return an array of sites: 
```
return [ { <site1 results object>, <site2 results object>, ..., <siteN results object>,]
```

The optional `extraData` property is meant to convey additional details which are noted on the site that might be helpful to the user. [<-- This needs more work, also probably should add notes on the optional `restrictions` property, not noted above.]

**Comment by tkjones117** [comment]
I think this would be a good addition to the README, if you don't mind opening a PR to add it.

**Comment by harcod** [comment]
Fixed by #216
